packages/server/src/fhircast/routes.ts

```diff
@@ -99,9 +101,39 @@ protectedCommonRoutes.post(
     }
 
     const topic = req.body['hub.topic'];
-    const subscriptionEndpoint = topic; // TODO: Create separate subscription endpoint for topic
-    const config = getConfig();
+    let subscriptionEndpoint: string;
+    try {
+      const results = await getRedis()
+        // Multi allows for multiple commands to be executed in a transaction
+        .multi()
+        // Sets the endpoint key for this topic if it doesn't exist
+        .setnx(`medplum:fhircast:topic:${topic}:endpoint`, generateId())
```

Comment:
I think `topic` is an unvalidated string at this point... Is there any validation we can do?  Will it always be a UUID?

Not sure what the risk is here, but feels kinda SQL-injection-ish.

Reply:
Topic can be any string as far as I know, but the scheme implemented should be "secure". In our case for SMART launch it is a UUID, but I don't know if we can enforce that.

https://fhircast.org/specification/STU2/#:~:text=MAY%20be%20a%20Universally%20Unique%20Identifier%20(UUID). 
> MAY be a Universally Unique Identifier ([UUID](https://tools.ietf.org/html/rfc4122)).

I think we should just recommend against using easily guessable topics

Reply:
Maybe a simple regex check for alphanumeric + dashes + underscores?  My primary concern here is a bad actor crafting a value that can muck with redis.

Reply:
Also should probably namespace this on project ID.  Especially if this is a user-submittable value, and we can probably assume users will do silly things like use very guessable `topic` like `abc` or whatever.

Reply:
> Maybe a simple regex check for alphanumeric + dashes + underscores? My primary concern here is a bad actor crafting a value that can muck with redis.

I'm ok with enforcing UUIDs or at least alpha + dashes + underscores. And maybe a minimum length

> Also should probably namespace this on project ID. Especially if this is a user-submittable value, and we can probably assume users will do silly things like use very guessable topic like abc or whatever.

We could make it connected to project, but then it would kill the use case of cross-project communication with a general purpose hub. Not sure how much we care about that but I did think that was one of the use cases